bundle/edu.gemini.p2checker/src/main/java/edu/gemini/p2checker/rules/general/GeneralRule.java

```diff
@@ -103,61 +102,49 @@ public IP2Problems check(ObservationElements elements)  {
-
-        private String formatGuiderString(final Iterator<GuideProbe> iter) {
-            if (iter == null || !iter.hasNext())
-                return "";
-
-            final StringBuilder buf = new StringBuilder();
-            buf.append(iter.next().getKey());
-            while (iter.hasNext()) {
-                buf.append(", ").append(iter.next().getKey());
-            }
-            return buf.toString();
+        private String formatGuiderString(final Collection<GuideProbe> probes) {
+            final String startDelim = probes.size() >= 2 ? "[" : "";
+            final String endDelim   = probes.size() >= 2 ? "]" : "";
+            final String sep        = ",";
+            final List<String> probeKeys = probes.stream().map(GuideProbe::getKey).collect(Collectors.toList());
+            return startDelim + String.join(sep, probeKeys) + endDelim;
         }
 
-        private void reportAltairLgsGuideIssues(P2Problems problems, Set<GuideProbe> guiders, AltairParams.Mode mode, ISPObsComponent targetComp) {
-            final ImList<GuideProbe> usedGuiders = mode.guiders();
-
-            // Check to see if we are using any guiders that are not in the set of guiders in the parameter list.
-            // Then check to see if there are any unused guiders in the set of guiders parameter list.
-            final ImList<GuideProbe> badGuiders = usedGuiders.filter(gp -> !guiders.contains(gp));
-            if (badGuiders.nonEmpty()) {
-                final String badGuiderNames = formatGuiderString(badGuiders.iterator());
-                problems.addError(PREFIX + "LGS_WFS", String.format(LGS_WFS, mode.displayValue(), badGuiderNames), targetComp);
-            } else if (guiders.size() - usedGuiders.size() > 1) {
-                final Set<GuideProbe> otherGuiders = new TreeSet<>(GuideProbe.KeyComparator.instance);
-                otherGuiders.addAll(guiders);
-                badGuiders.foreach(otherGuiders::remove);
+        private void reportAltairLgsGuideIssues(final P2Problems problems, final Set<GuideProbe> usedGuiders,
+                                                final AltairParams.Mode mode, final ISPObsComponent targetComp) {
+            // The "bad guiders", i.e. the used guiders that cannot be used with the mode.
+            final Set<GuideProbe> badGuiders = new TreeSet<>(GuideProbe.KeyComparator.instance);
+            badGuiders.addAll(usedGuiders);
+            mode.guiders().foreach(badGuiders::remove);
+
+            // We must be using the TTF guider, which is the head of the mode guider list.
+            final GuideProbe ttfGuider = mode.guiders().head();
+            if (!usedGuiders.contains(ttfGuider)) {
+                problems.addError(PREFIX + "LGS_WFS", String.format(LGS_WFS, mode.displayValue(), ttfGuider), targetComp);
+            } else if (!badGuiders.isEmpty()) {
+                // The good guiders, i.e. the ones that can be used with the mode and are in use.
```

Comment:
This only happens if guiders are in use that shouldn't be.
We don't allow this in the new implementation, I don't think, so it should never come up except possibly in old programs.